pkg/result/ignore.go

```diff
@@ -0,0 +1,171 @@
+package result
+
+import (
+	"bufio"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+	"golang.org/x/exp/slices"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+// IgnoreFinding represents an item to be ignored.
+type IgnoreFinding struct {
+	// ID is the identifier of the vulnerability, misconfiguration, secret, or license.
+	// e.g. CVE-2019-8331, AVD-AWS-0175, etc.
+	// required: true
+	ID string `yaml:"id"`
+
+	// Paths is the list of file paths to be ignored.
+	// If Paths is not set, the ignore finding is applied to all files.
+	// required: false
+	Paths []string `yaml:"paths"`
+
+	// ExpiredAt is the expiration date of the ignore finding.
+	// If ExpiredAt is not set, the ignore finding is always valid.
+	// required: false
+	ExpiredAt time.Time `yaml:"expired_at"`
+
+	// Statement describes the reason for ignoring the finding.
+	// required: false
+	Statement string `yaml:"statement"`
```

Comment:
yaml files support comments. 
Do you think we need this field?

e.g.:
```yaml
vulnerabilities:
  # Accept the risk
  - id: CVE-2023-30861
    paths:
      - "pkg/fanal/analyzer/language/python/poetry/testdata/happy/poetry.lock"
      - "pkg/fanal/analyzer/language/python/pip/testdata/requirements.txt"
```

Reply:
Yes, we want to generate VEX from .trivyignore.yaml in the future. VEX has `impact_statement`, so we should have a corresponding field.
https://github.com/openvex/spec/blob/main/OPENVEX-SPEC.md#statement-fields